### PR TITLE
Log return code of hook command

### PR DESCRIPTION
**Pull request by perlpunk**
Issue: https://progress.opensuse.org/issues/99519

### CODE REVIEW

File: lib/OpenQA/Task/Job/FinalizeResults.pm [hunk 1]
```diff
@@ -53,7 +53,8 @@ sub _finalize_results {
         my $timeout      = $ENV{OPENQA_JOB_DONE_HOOK_TIMEOUT}      // '5m';
         my $kill_timeout = $ENV{OPENQA_JOB_DONE_HOOK_KILL_TIMEOUT} // '30s';
         my $ret          = _done_hook_new_issue($openqa_job, $hook, $timeout, $kill_timeout);
-        $minion_job->note(hook_cmd => $hook, hook_result => $ret);
+        my $rc           = $?;
+        $minion_job->note(hook_cmd => $hook, hook_result => $ret, rc => $rc);
```
Comment (Martchus) [inline lib/OpenQA/Task/Job/FinalizeResults.pm:57]:
```suggestion
        $minion_job->note(hook_cmd => $hook, hook_result => $ret, hook_rc => $rc);
```

**Review by Martchus — COMMENTED** [review]
LGTM, just one suggestion.

File: lib/OpenQA/Task/Job/FinalizeResults.pm [hunk 2]
```diff
@@ -53,7 +53,8 @@ sub _finalize_results {
         my $timeout      = $ENV{OPENQA_JOB_DONE_HOOK_TIMEOUT}      // '5m';
         my $kill_timeout = $ENV{OPENQA_JOB_DONE_HOOK_KILL_TIMEOUT} // '30s';
         my $ret          = _done_hook_new_issue($openqa_job, $hook, $timeout, $kill_timeout);
-        $minion_job->note(hook_cmd => $hook, hook_result => $ret);
+        my $rc           = $?;
```
Comment (okurz) [inline lib/OpenQA/Task/Job/FinalizeResults.pm:56]: Does the `$?` actually represent the exit code of the hook command? How did you test? Please describe in the commit

**Review by okurz — CHANGES_REQUESTED** [review]

File: lib/OpenQA/Task/Job/FinalizeResults.pm (hunk 1 above)
Comment (perlpunk) [reply lib/OpenQA/Task/Job/FinalizeResults.pm]: done

File: lib/OpenQA/Task/Job/FinalizeResults.pm (hunk 2 above)
Comment (perlpunk) [reply lib/OpenQA/Task/Job/FinalizeResults.pm]: I added a test for hook_rc in 10-jobs.t now.
`$?` contains the exit code of the last command, e.g. `qx{}` or `system()`.
I rearranged it a bit now, to make sure `$?` is used directly after the call.

**Review by Martchus — APPROVED** [review]

File: lib/OpenQA/Task/Job/FinalizeResults.pm (hunk 2 above)
Comment (perlpunk) [reply lib/OpenQA/Task/Job/FinalizeResults.pm]: I also added an error case to check if we get the correct exit code

**Review by kalikiana — APPROVED** [review]

File: lib/OpenQA/Task/Job/FinalizeResults.pm (hunk 2 above)
Comment (okurz) [reply lib/OpenQA/Task/Job/FinalizeResults.pm]: very nice. The test explains it well enough

**Review by okurz — APPROVED** [review]

**Comment by okurz** [comment]
recorded the flaky coverage in https://progress.opensuse.org/issues/99597 and https://progress.opensuse.org/issues/99594

**Pull request merged**